qiskit-superstaq/qiskit_superstaq/superstaq_job.py

```diff
@@ -300,6 +300,57 @@ def input_circuits(
         """
         return self._get_circuits("input_circuit", index)
 
+    @overload
+    def pulse_gate_circuits(self, index: int) -> qiskit.QuantumCircuit:
+        ...
+
+    @overload
+    def pulse_gate_circuits(self, index: None = None) -> list[qiskit.QuantumCircuit]:
+        ...
+
+    def pulse_gate_circuits(
```

Comment:
out of curiosity how complicated would it be to have this use `_get_circuits`? it seems like it'd have to be updated to throw an error (instead of refreshing) if  `circuit_type` is in `_job_info[job_id]` but its value is None, but then it would avoid some duplication

Reply:
How do you  feel about moving that check into get_circuits? I guess a weird thing is it only applies to pulse_gate_circuits but in theory it could apply to other circuits that are only supported by specific devices in the future

Reply:
yeah i think that's fine/the way you did it seems reasonable to me!